include/viz/ParticleSystem.hpp

```diff
@@ -62,26 +68,77 @@ class ParticleSystem : public sf::Drawable
 			auto new_pos = p.getPosition() + additional_displacement;
 
 			// make sure particles don't escape the rect
-			if (new_pos.x >= (rect.position.x + rect.size.x))
-				// teleport from right edge to left
-				new_pos.x = rect.position.x + -p.getRadius();
-			else if (new_pos.x + p.getRadius() < 0)
-				// teleport from left edge to right
-				new_pos.x = (rect.position.x + rect.size.x);
+
+			switch (start_position)
+			{
+			case StartSide::BOTTOM:
+			case StartSide::TOP:
+				if (new_pos.x >= (rect.position.x + rect.size.x))
+					// teleport from right edge to left
+					new_pos.x = rect.position.x + -p.getRadius();
+				else if (new_pos.x + p.getRadius() < 0)
+					// teleport from left edge to right
+					new_pos.x = (rect.position.x + rect.size.x);
+				break;
+			case StartSide::LEFT:
+			case StartSide::RIGHT:
+				if (new_pos.y >= (rect.position.y + rect.size.y))
+					// teleport from right edge to left
```

Comment:
it would be nice if these comments described whether we are going from top-to-bottom or bottom-to-top
```suggestion
					// teleport from top edge to bottom
```